File: Examples/CSharp/ProgrammersGuide/Working-Shapes/ChangeShapeSize.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSharp.ProgrammersGuide.Working_Shapes
{
    public class ChangeShapeSize
    {
        public static void Run()
        {
            //ExStart:ChangeShapeSize
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();

            // call a Diagram class constructor to load the VSDX diagram
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");
            // get page by name
            Page page = diagram.Pages.GetPage("Page-1");
            // get shape by id
            Shape shape = page.Shapes.GetShape(796);
            // alter the size of Shape
            shape.SetWidth(2 * shape.XForm.Width.Value);
            shape.SetHeight(2 * shape.XForm.Height.Value);
            // save diagram
            diagram.Save(dataDir + "ChangeShapeSize_Out.vsdx", SaveFileFormat.VSDX);
            //ExEnd:ChangeShapeSize
        }
    }
}

File: Examples/VisualBasic/Working-Shapes/ChangeShapeSize.vb
Imports Aspose.Diagram
Imports System
Public Class ChangeShapeSize
    Public Shared Sub Run()
        ' ExStart:ChangeShapeSize
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_Shapes()

        ' Call a Diagram class constructor to load the VSDX diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))
        ' Get page by name
        Dim page As Page = diagram.Pages.GetPage("Page-1")
        ' Get shape by id
        Dim shape As Shape = page.Shapes.GetShape(796)
        ' Alter the size of Shape
        shape.SetWidth(2 * shape.XForm.Width.Value)
        shape.SetHeight(2 * shape.XForm.Height.Value)
        ' Save diagram
        diagram.Save(dataDir & Convert.ToString("ChangeShapeSize_out.vsdx"), SaveFileFormat.VSDX)
        ' ExEnd:ChangeShapeSize
    End Sub
End Class
